Sources/Subprocess/Platforms/Subprocess+Darwin.swift

```diff
@@ -477,26 +477,41 @@ extension Configuration {
     }
 }
 
-// Special keys used in Error's user dictionary
-extension String {
-    static let debugDescriptionErrorKey = "NSDebugDescription"
+// MARK: - ProcessIdentifier
+
+/// A platform independent identifier for a Subprocess.
+public struct ProcessIdentifier: Sendable, Hashable {
```

Comment:
Could you make this type move-only and incorporate the `close()` operation into `deinit`?

Reply:
Closing might involve closing FDs right? Which might be an asynchronous and throwing operation.

Reply:
We've had this discussion in another place ([here?](https://github.com/swiftlang/swift-subprocess/blob/5715ed49b0a5493cb24f3904dc2d9736c180d949/Sources/Subprocess/Configuration.swift#L689)). We ended up not calling `close`, but asserting that `close` has already been called in `deinit`.

But it does seem like a design people are going to reach for repeatedly. I wonder if we can put our thought process down somewhere.

Reply:
ProcessIdentifier is only ever provided as a parameter to a closure called within a with-style function, except for runDetached. If we remove the latter from the API, there's no place where ProcessIdentifier needs to be used as a handle that needs to be closed and therefore no reason it needs to be responsible for lifetime management.